### PR TITLE
working version testsuite databases

### PR DESCRIPTION
**Pull request by paulineml**
Adding the hack zurich databse to let model train. 
Changes to also see the views from the table
Changes to enable numeric keyword for sql queries

### CODE REVIEW

File: src/spider/test_suite_eval/evaluation.py [hunk 1]
```diff
@@ -575,7 +575,12 @@ def evaluate(gold, predict, db_dir, etype, kmaps, plug_value, keep_distinct, pro
             p_str = p_str.replace("value", "1")
             g_str, db, question = g
             db_name = db
-            db = os.path.join(db_dir, db, db + ".db")
+            if db_name == 'hack_zurich':
```
Comment (delixfe) [inline src/spider/test_suite_eval/evaluation.py:578]: We could also rename the hackzurich db to `hack_zurich.sqlite`. Then we would not need the if case here.

Comment (paulineml) [reply src/spider/test_suite_eval/evaluation.py:578]: Yes, but I have issues when I try to convert the .db to .sqlite, I will try again now but it will take me some more time.

Comment (brunnurs) [reply src/spider/test_suite_eval/evaluation.py]: Hi all,

Quick comment from my side: I would not recommend changing your database to an sqlite-database at that point. It is also not really necessary, let me quickly explain what the problem is:

After each epoch of training the main training loop https://github.com/brunnurs/valuenet/blob/1ce7e56956b378a8f281e9f9919e6aa98516a9d9/src/main.py#L85
will call evaluation on the spider *DEV* split. This was the default so far even when working with other datasets. As we always concatenate all training data (so Spider data and e.g. the hack zurich data ), the evaluation on the spider DEV set is always a good proxy to see what the model learned. Obviously the best evaluation metric would be to evaluate the model on **all** DEV data after each epoch. I will quickly look into this and let you know if we can implement that until Friday. Alternatively we can also just  remove this part: https://github.com/brunnurs/valuenet/blob/1ce7e56956b378a8f281e9f9919e6aa98516a9d9/src/main.py#L85-L94
and everything should work fine again. You still have an evaluation metric at that point, `acc` and `sketch_acc` which tell you how how many samples of the *DEV* set were correct - just not as fine grained as the spider_evaluation.

Comment (brunnurs) [reply src/spider/test_suite_eval/evaluation.py]: I added the new config flag `--run_spider_evaluation_after_epoch`  to exclude spider evaluation by default. I tested the training without it and it works fine, just reporting `accuracy` and `sketch_accuracy` now.

A quick explanation on this two metrics:
* **accuracy** compares the predicted abstract query tree (so the actually query in SemQL) with the ground truth.
* **sketch_accuracy** is only comparing the logic form of a query, without verifying if the correct tables/columns/values were selected.

For you I guess only **accuracy** is interesting.
